src/Type.cc

```diff
@@ -1205,7 +1245,12 @@ void RecordType::AddField(unsigned int field, const TypeDecl* td)
 		TypeTag tag = type->Tag();
 
 		if ( tag == TYPE_RECORD )
-			init = std::make_unique<detail::RecordFieldInit>(cast_intrusive<RecordType>(type));
+			{
+			// Initially, put record fields into creation_inits. Once parsing has
+			// completed, they may move into deferred_inits. See DoneParsing().
+			auto rfi = std::make_unique<detail::RecordFieldInit>(cast_intrusive<RecordType>(type));
+			creation_inits.emplace_back(field, std::move(rfi));
+			}
```

Comment:
I believe it's ok/expected to leave `init` unset here, correct? Want to add comment saying so?

Reply:
Yep, it's the same pattern as for ExpressionFieldInit `efi` above. I've added a comment.